sdk/identity/identity/src/credentials/interactiveBrowserCredentialOptions.ts

```diff
@@ -25,76 +25,71 @@ export type InteractiveBrowserAuthenticationFlow = "implicit-grant" | "auth-code
-
-  /**
-   * The Azure Active Directory tenant (directory) ID.
-   */
-  tenantId?: string;
+export type InteractiveBrowserCredentialOptions = TokenCredentialOptions &
+  InteractiveCredentialOptions & {
+    /**
+     * Gets the redirect URI of the application. This should be same as the value
+     * in the application registration portal.  Defaults to `window.location.href`.
+     */
+    redirectUri?: string | (() => string);
 
-  /**
-   * Result of a previous authentication that can be used to retrieve the cached credentials of each individual account.
-   * This is necessary to provide in case the application wants to work with more than one account per
-   * Client ID and Tenant ID pair.
-   *
-   * This record can be retrieved by calling to the InteractiveBrowserCredential's `authenticate()` method, as follows:
-   *
-   *     const authenticationRecord = await credential.authenticate();
-   *
-   */
-  authenticationRecord?: AuthenticationRecord;
+    /**
+     * Gets the URI to which the user will be redirected when logging out.
+     * Defaults to `window.location.href`.
+     */
+    postLogoutRedirectUri?: string | (() => string);
 
-  /**
-   * Disables the automatic request of user interaction when the token is invalid,
-   * forcing developers to use the `authenticate()` method of the credential to
-   * trigger the user interaction and then refresh the token.
-   */
-  disableAutomaticAuthentication?: boolean;
-}
+    /**
+     * The Azure Active Directory tenant (directory) ID.
+     */
+    tenantId?: string;
 
-/**
- * Defines options for the InteractiveBrowserCredential class for NodeJS.
- */
-export type InteractiveBrowserCredentialOptions = InteractiveBrowserCredentialCommonOptions &
-  InteractiveCredentialOptions & {
     /**
      * The client (application) ID of an App Registration in the tenant.
      */
     clientId?: string;
   };
 
 /**
- * Defines options for the InteractiveBrowserCredential class for the browser.
+ * Defines the common options for the InteractiveBrowserCredential class.
  */
-export interface InteractiveBrowserCredentialBrowserOptions
-  extends InteractiveBrowserCredentialCommonOptions {
-  /**
-   * The client (application) ID of an App Registration in the tenant.
-   * This parameter is required on the browser.
-   */
-  clientId: string;
-  /**
-   * Specifies whether a redirect or a popup window should be used to
-   * initiate the user authentication flow. Possible values are "redirect"
-   * or "popup" (default) for browser and "popup" (default) for node.
-   *
-   */
-  loginStyle?: BrowserLoginStyle;
-  /**
-   * Authentication flow to use.
-   * If the user specifies the implicit-grant flow, we will use MSAL 1.
-   * Otherwise, auth-code will be assumed, which uses PKCE and MSAL 2.
-   */
-  flow?: InteractiveBrowserAuthenticationFlow;
-}
+export type InteractiveBrowserCredentialBrowserOptions = TokenCredentialOptions &
+  InteractiveCredentialOptions & {
```

Comment:
Could this be simplified to 
```typescript
export type InteractiveBrowserCredentialBrowserOptions = InteractiveBrowserCredentialOptions & {
  loginStyle?: BrowserLoginStyle;
  flow?: InteractiveBrowserAuthenticationFlow;
};
```

Reply:
well, the issue is that in the browser one, `clientId: string` is required, since we can't use the Azure CLI client ID 🤔 my attempt was to keep them separate to remove the extra intermediate interface that represents the common properties. In the browser I am throwing if `clientId` is not provided anyway, so I could make it optional in the options, but I'd rather give as much help as possible to the users.